### ISSUE TITLE
ECS: Cannot use environment variable interpolation in AWS auto scaling

### ISSUE THREAD

**Issue by parisk** [docker/compose-cli#1580]
**Description**

Cannot use environment variable interpolation in AWS auto scaling.

**Steps to reproduce the issue:**
1. Create a simple `docker-compose.yml` with environment interpolated auto scaling:
    ```yml
    version: "3.8"

    services:
      web:
        image: nginx
        deploy:
          x-aws-autoscaling:
            min: ${AUTO_SCALING_MIN:-1}
            max: ${AUTO_SCALING_MAX:-5}
            cpu: 60
    ```
2. Run `docker compose up` or `docker compose convert`

**Describe the results you received:**

```
% docker --context ecs compose convert
json: cannot unmarshal string into Go struct field autoscalingConfig.max of type int
```

**Describe the results you expected:**

Expected the stack to be deployed or converted to CloudFormation respectively.

**Output of `docker version`:**

```
Client: Docker Engine - Community
 Cloud integration: 1.0.12
 Version:           20.10.5
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        55c4c88
 Built:             Tue Mar  2 20:13:00 2021
 OS/Arch:           darwin/amd64 (rosetta)
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.5
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       363e9a8
  Built:            Tue Mar  2 20:16:48 2021
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.4.4
  GitCommit:        05f951a3781f4f2c1911b05e61c160e9c30eaa8e
 runc:
  Version:          1.0.0-rc93
  GitCommit:        12644e614e25b05da6fd08a38ffa0cfe1903fdec
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Output of `docker context show`:**  

```
[
    {
        "Name": "ecs",
        "Metadata": {
            "Type": "ecs"
        },
        "Endpoints": {
            "docker": {
                "SkipTLSVerify": false
            },
            "ecs": {
                "Profile": "sl"
            }
        },
        "TLSMaterial": {},
        "Storage": {
            "MetadataPath": "/Users/paris/.docker/contexts/meta/b7fc8f0bb58273286f4068dc9c6aeddf2c87fc299fdd5e82898523b6c6f5cbab",
            "TLSPath": "/Users/paris/.docker/contexts/tls/b7fc8f0bb58273286f4068dc9c6aeddf2c87fc299fdd5e82898523b6c6f5cbab"
        }
    }
]

```

**Output of `docker info`:**

```
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  app: Docker App (Docker Inc., v0.9.1-beta3)
  buildx: Build with BuildKit (Docker Inc., v0.5.1-docker)
  scan: Docker Scan (Docker Inc., v0.6.0)

Server:
 Containers: 38
  Running: 17
  Paused: 0
  Stopped: 21
 Images: 41
 Server Version: 20.10.5
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 05f951a3781f4f2c1911b05e61c160e9c30eaa8e
 runc version: 12644e614e25b05da6fd08a38ffa0cfe1903fdec
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 5.10.25-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: aarch64
 CPUs: 4
 Total Memory: 7.765GiB
 Name: docker-desktop
 ID: DLFE:QHL3:IPHL:6JAY:IMTK:T4YU:6LQG:MJA2:N6SM:3QXZ:SHUI:PXCV
 Docker Root Dir: /var/lib/docker
 Debug Mode: true
  File Descriptors: 66
  Goroutines: 74
  System Time: 2021-04-21T10:53:08.496978087Z
  EventsListeners: 3
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

**Comment by ndeloof** [comment]
I guess compose-go interpolation only applies to spec-defined fields, but not to extensions with nested structure

**Comment by yordis** [comment]
I am having issues where `docker-compose`  don't read the HOST environment variables anymore

**Comment by yordis** [comment]
I had to convert my environment injection to use `${ENV}` syntax rather than picking the OS environment directly

**Comment by ndeloof** [comment]
Support for int attribute interpolation has been fixed by https://github.com/compose-spec/compose-go/pull/142

**Comment by slavdok** [comment]
> Support for int attribute interpolation has been fixed by [compose-spec/compose-go#142](https://github.com/compose-spec/compose-go/pull/142)

What and where am I supposed to update to fix this issue? This was marked fixed in 2021. I've installed Docker and AWS CLI from scratch in 2022. I am having this exact same issue as described.

Server: Docker Desktop 4.12.0 (85629)
 Engine:
  Version:          20.10.17
Docker Compose version v2.10.2
